packages/wp-dashboard/src/components/editorSettings/utils/validateShopifyHost.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export default function isValidShopifyHost(value = '') {
+  const startWith = /https?:\/\/?/;
+  if (Boolean(value.toLowerCase().match(startWith)) || value.includes('www.')) {
+    return false;
+  }
+  return true;
```

Comment:
Interesting combination of regex and string search.

I'd probably simply have used something like `/[\d\w-]+\.myshopify\.com/i`

WDYT?

Reply:
I was trying to write a Regex to handle "all the cases" but my Regex isn't the best ... happy to use just a Regex for sure.

That said

> By default, your online store has a myshopify.com

I was going to validate using myshopify.com but that's only the default 

https://help.shopify.com/en/manual/domains

>  If you want to change the web address that's displayed to customers while they browse your online store, then you need to use a custom domain

So really we should just be looking for a domain without http(s):// and no www correct?

Reply:
AFAIK Even if you change your shop‘s domain, the one we need for the API always is on myshopify.com working.
We should enforce that format.

Reply:
See an example here: https://shopify.dev/api/admin/getting-started

The API endpoint is not available under the custom domain.